internal/services/formatters/go/gosec/formatter_test.go

```diff
@@ -24,112 +24,184 @@ import (
 
 	"github.com/ZupIT/horusec-devkit/pkg/entities/analysis"
 	"github.com/ZupIT/horusec-devkit/pkg/enums/tools"
-	cliConfig "github.com/ZupIT/horusec/config"
-	"github.com/ZupIT/horusec/internal/entities/workdir"
+	"github.com/ZupIT/horusec/config"
 	"github.com/ZupIT/horusec/internal/services/docker"
 	"github.com/ZupIT/horusec/internal/services/formatters"
 )
 
-func TestGoLang_StartAnalysis(t *testing.T) {
-	t.Run("Should run analysis and return error because output is empty", func(t *testing.T) {
-		dockerAPIControllerMock := &docker.Mock{}
-		dockerAPIControllerMock.On("SetAnalysisID")
-		dockerAPIControllerMock.On("CreateLanguageAnalysisContainer").Return("", nil)
+func TestGosecStartAnalysis(t *testing.T) {
+	t.Run("Should run analysis with sucessfull when output is empty", func(t *testing.T) {
+		docker := &docker.Mock{}
+		docker.On("SetAnalysisID")
+		docker.On("CreateLanguageAnalysisContainer").Return("", nil)
 
-		config := &cliConfig.Config{}
-		config.WorkDir = &workdir.WorkDir{}
+		cfg := config.New()
 
-		service := formatters.NewFormatterService(&analysis.Analysis{}, dockerAPIControllerMock, config)
+		entity := new(analysis.Analysis)
+		service := formatters.NewFormatterService(entity, docker, cfg)
 
-		golangAnalyzer := NewFormatter(service)
-
-		assert.NotPanics(t, func() {
-			golangAnalyzer.StartAnalysis("")
-		})
+		NewFormatter(service).StartAnalysis("")
+		assert.False(t, entity.HasErrors(), "Expected no error for analysis")
 	})
 
-	t.Run("Should run analysis without panics and save on cache with success", func(t *testing.T) {
-		dockerAPIControllerMock := &docker.Mock{}
-		dockerAPIControllerMock.On("SetAnalysisID")
+	t.Run("Should run analysis with successful and add vulnerability on Analysis", func(t *testing.T) {
```

Comment:
Same here `Should run analysis successfully and add vulnerability on Analysis`.

Reply:
Done